File: JavaScript/6 kyu/6 kyu - Replace With Alphabet Position.js
// https://www.codewars.com/kata/replace-with-alphabet-position/train/javascript
// My solution
function alphabetPosition(text) {
    return !text.match("[a-zA-Z]") ? "" : text
        .toLowerCase()
        .match(/[a-z]/g)
        .map(ch => ch.charCodeAt(0) - 96)
        .join(" ")
}

// ...
function alphabetPosition(text) {
    return text
        .toUpperCase()
        .match(/[a-z]/gi)
        .map( (c) => c.charCodeAt() - 64)
        .join(' ');
}

// ...
let alphabetPosition = (text) => text
    .toUpperCase()
    .replace(/[^A-Z]/g, '')
    .split('')
    .map(ch => ch.charCodeAt(0) - 64)
    .join(' ');

File: Python/6 - kyu/6 kyu - Replace With Alphabet Position.py
# https://www.codewars.com/kata/replace-with-alphabet-position/train/python
# My solution
def alphabet_position(text):
    letters = filter(str.isalpha, text.lower())
    nth_letters = map(lambda ch: ord(ch) - ord('a') + 1, letters)
    return " ".join(map(str, nth_letters))

# ...
def alphabet_position(text):
    return ' '.join(str(ord(c) - 96) for c in text.lower() if c.isalpha())